src/genconstants.c

```diff
@@ -0,0 +1,296 @@
+#include <curses.h>
+
+#define PCONST(ty, NAME) printf("pub const " #NAME ": " #ty " = %lld;\n", (long long) (NAME))
+#define PCONSTU(ty, NAME) printf("pub const " #NAME ": " #ty " = %llu;\n", (unsigned long long) (NAME))
+
+int main() {
+        /* some values aren't set until after this is run */
+        printf("//");
+	initscr();
+	endwin();
+        printf("\n");
+
+	/* Success/Failure. */
+	PCONST(i32, ERR);
+	PCONST(i32, OK);
+	PCONST(c_bool, TRUE);
+	PCONST(c_bool, FALSE);
+
+	/* Attributes. */
+#ifdef NCURSES_ATTR_SHIFT
+	PCONST(u32, NCURSES_ATTR_SHIFT);
+#endif
+
+	/* Colors */
+	PCONST(i16, COLOR_BLACK);
+	PCONST(i16, COLOR_RED);
+	PCONST(i16, COLOR_GREEN);
+	PCONST(i16, COLOR_YELLOW);
+	PCONST(i16, COLOR_BLUE);
+	PCONST(i16, COLOR_MAGENTA);
+	PCONST(i16, COLOR_CYAN);
+	PCONST(i16, COLOR_WHITE);
+
+	/* Values for the _flags member */
+#ifdef _SUBWIN
+	PCONST(i32, _SUBWIN);
+#endif
+
+#ifdef _ENDLINE
+	PCONST(i32, _ENDLINE);
+#endif
+
+#ifdef _FULLWIN
+	PCONST(i32, _FULLWIN);
+#endif
+
+#ifdef _SCROLLWIN
+	PCONST(i32, _SCROLLWIN);
+#endif
+
+#ifdef _ISPAD
+	PCONST(i32, _ISPAD);
+#endif
+
+#ifdef _HASMOVED
+	PCONST(i32, _HASMOVED);
+#endif
+
+#ifdef _WRAPPED
+	PCONST(i32, _WRAPPED);
+#endif
+
+	/*
+	 * This value is used in the firstchar and lastchar fields to mark
+	 * unchanged lines
+	 */
+#ifdef _NOCHANGE
+	PCONST(i32, _NOCHANGE);
+#endif
+
+	/*
+	 * This value is used in the oldindex field to mark lines created by insertions
+	 * and scrolls.
+	 */
+#ifdef _NEWINDEX
+	PCONST(i32, _NEWINDEX);
+#endif
+
+	/* Keys */
+	PCONST(i32, KEY_CODE_YES);
+	PCONST(i32, KEY_MIN);
+	PCONST(i32, KEY_BREAK);
+	PCONST(i32, KEY_SRESET);
+	PCONST(i32, KEY_RESET);
+	PCONST(i32, KEY_DOWN);
+	PCONST(i32, KEY_UP);
+	PCONST(i32, KEY_LEFT);
+	PCONST(i32, KEY_RIGHT);
+	PCONST(i32, KEY_HOME);
+	PCONST(i32, KEY_BACKSPACE);
+	PCONST(i32, KEY_F0);
+	PCONST(i32, KEY_DL);
+	PCONST(i32, KEY_IL);
+	PCONST(i32, KEY_DC);
+	PCONST(i32, KEY_IC);
+	PCONST(i32, KEY_EIC);
+	PCONST(i32, KEY_CLEAR);
+	PCONST(i32, KEY_EOS);
+	PCONST(i32, KEY_EOL);
+	PCONST(i32, KEY_SF);
+	PCONST(i32, KEY_SR);
+	PCONST(i32, KEY_NPAGE);
+	PCONST(i32, KEY_PPAGE);
+	PCONST(i32, KEY_STAB);
+	PCONST(i32, KEY_CTAB);
+	PCONST(i32, KEY_CATAB);
+	PCONST(i32, KEY_ENTER);
+	PCONST(i32, KEY_PRINT);
+	PCONST(i32, KEY_LL);
+#ifdef A1
+	PCONST(i32, A1);
+#endif
+#ifdef A3
+	PCONST(i32, A3);
+#endif
+#ifdef B2
+	PCONST(i32, B2);
+#endif
+#ifdef C1
+	PCONST(i32, C1);
+#endif
+#ifdef C3
+	PCONST(i32, C3);
```

Comment:
Are these `#ifdef` supposed to be prefixed by `KEY_` ? because I'm missing at least `KEY_B2`